tests/unit/test_login_manager.py

```diff
@@ -0,0 +1,116 @@
+import re
+from unittest.mock import patch
+
+import click
+import pytest
+
+from globus_cli.login_manager import requires_login
+
+
+def mock_get_tokens(resource_server):
+    fake_tokens = {
+        "a.globus.org": {
+            "access_token": "fake_a_access_token",
+            "refresh_token": "fake_a_refresh_token",
+        },
+        "b.globus.org": {
+            "access_token": "fake_b_access_token",
+            "refresh_token": "fake_b_refresh_token",
+        },
+    }
+
+    return fake_tokens.get(resource_server)
+
+
+@patch("globus_cli.tokenstore.token_storage_adapter")
+def test_requires_login_success(mock_get_adapter):
+    mock_get_adapter._instance.get_token_data = mock_get_tokens
+
+    # single server
+    @requires_login("a.globus.org")
+    def dummy_command():
+        return True
+
+    assert dummy_command()
```

Comment:
Some would, perhaps rightfully, dispute the legibility of this, but I thought I would offer the suggestion since it would shorten a lot of tests.
```suggestion
    assert requires_login("a.globus.org")(lambda: True)()
```

Reply:
I know they're logically equivalent, but I wanted to make sure the @ decorator usage was functioning as intended since that's how we'll be calling this in code

Reply:
This is the sort of thing which I probably wouldn't bother to call out one way or another in most cases. Either syntax is fine and they mean the same thing.

If you had half a dozen variants of the same decorator being applied to `lambda: True` equivalents, inlining them like this would possibly make the test more readable, but that's not the case here.

I bias slightly more towards readability than brevity, so I'm going to mark this as resolved.